### PR TITLE
solution for Registration Service

### PR DESCRIPTION
**Pull request by vladis9**

### CODE REVIEW

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 1]
```diff
@@ -5,10 +5,34 @@
...
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RuntimeException("Data must be filled");
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:16]: It is better to create your own exception, for example RegistrationException

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 2]
```diff
@@ -5,10 +5,34 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
```
Comment (kozhukhovsky) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:14]: add check
image

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin("Abracadabra");
+        user.setAge(20);
+        user.setPassword("arbadacarbA");
+    }
+
+    @Test
+    void valid_user_Ok() {
```
Comment (kozhukhovsky) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:29]: rename all method with this pattern
image

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
```
Comment (kozhukhovsky) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:11]: add new test for check with same login

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]

**Review by kozhukhovsky — CHANGES_REQUESTED** [review]
For platform

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+        user.setPassword("password9475");
+        user.setAge(null);
+
+        RegistrationException exception = Assertions.assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        Assertions.assertEquals("User must be at least 18 years old", exception.getMessage());
+    }
+
+    @Test
+    public void register_underAge_notOk() {
```
Comment (okuzan) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:118]:
```suggestion
    public void register_underageUser_notOk() {
```

**Review by okuzan — CHANGES_REQUESTED** [review]
You should test edge cases
image

**Pull request closed without merging**